src/server/server.ts

```diff
@@ -522,6 +522,9 @@ namespace ts.server {
                 return;
             }
 
+            // removed = deleted ? true : (added ? false : undefined)

```

Comment:
It was intended as an explanation.  If you don't believe it's helpful, I'm happy to remove it.